PandorasBox/Features/Actions/AutoCordial.cs

```diff
@@ -0,0 +1,135 @@
+using PandorasBox.Features;
+using PandorasBox.FeaturesSetup;
+using ECommons.DalamudServices;
+using FFXIVClientStructs.FFXIV.Client.Game;
+using FFXIVClientStructs.FFXIV.Client.Game.UI;
+using Dalamud.Game;
+using Dalamud.Game.ClientState.Conditions;
+using System.Linq;
+using ECommons.Throttlers;
+using ImGuiNET;
+using System.Collections.Generic;
+
+namespace PandorasBox.Features.Actions
+{
+    public unsafe class AutoCordial : Feature
+    {
+        public override string Name => "Auto-Cordial";
+
+        public override string Description => "Automatically use a cordial when below the threshold.";
+
+        public override FeatureType FeatureType => FeatureType.Actions;
+
+        public Configs Config { get; private set; }
+
+        public override bool UseAutoConfig => false;
+
+        public class Configs : FeatureConfig
+        {
+            [FeatureConfigOption("GP Threshold", IntMin = 0, IntMax = 1000, EditorSize = 300)]
+            public int DefaultThreshold = 1;
+            public bool DirectionAbove = true;
+            public bool DirectionBelow = false;
+
+            [FeatureConfigOption("Invert Priority (Watered -> Regular -> Hi)")]
+            public bool InvertPriority = false;
+
+            [FeatureConfigOption("Prevent Overcap")]
+            public bool PreventOvercap = true;
+
+            [FeatureConfigOption("Use on Fisher")]
+            public bool UseOnFisher = false;
+        }
+
+        protected override DrawConfigDelegate DrawConfigTree => (ref bool _) =>
+        {
+            ImGui.PushItemWidth(300);
+            ImGui.SliderInt("GP Threshold", ref Config.DefaultThreshold, 1, 1000);
+            if (ImGui.RadioButton("Above", Config.DirectionAbove))
+            {
+                Config.DirectionAbove = true;
+                Config.DirectionBelow = false;
+            }
+            ImGui.SameLine();
+            if (ImGui.RadioButton("Below", Config.DirectionBelow))
+            {
+                Config.DirectionAbove = false;
+                Config.DirectionBelow = true;
+            }
+            ImGui.Checkbox("Invert Priority (Watered -> Regular -> Hi)", ref Config.InvertPriority);
+            ImGui.Checkbox("Prevent Overcap", ref Config.PreventOvercap);
+            ImGui.Checkbox("Use on Fisher", ref Config.UseOnFisher);
+        };
+
+        private bool WillOvercap(int gp_recovery)
+        {
+            return Svc.ClientState.LocalPlayer.CurrentGp + gp_recovery < Svc.ClientState.LocalPlayer.MaxGp;
+        }
+
+        private void RunFeature(Framework framework)
+        {
+            if (!(Svc.ClientState.LocalPlayer.ClassJob.Id == 16) || !(Svc.ClientState.LocalPlayer.ClassJob.Id == 17)) return;
```

Comment:
Add null check for LocalPlayer before this

Reply:
done in 1e66a08